pkg/asset/manifests/openshift.go

```diff
@@ -214,7 +214,9 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 		assetData["99_baremetal-provisioning-config.yaml"] = applyTemplateData(baremetalConfig.Files()[0].Data, bmTemplateData)
 	}
 
-	if platform == azuretypes.Name && installConfig.Config.Publish == types.InternalPublishingStrategy {
+	if platform == azuretypes.Name &&
+		installConfig.Config.Publish == types.InternalPublishingStrategy &&
+		installConfig.Config.Azure.OutboundType == azuretypes.LoadbalancerOutboundType {
```

Comment:
In what configuration this outbound service is still needed? Can it be removed?

Reply:
@mjudeikis  It is required for internal clusters with outbound-loadbalancer. We cannot depend on machinesets to manage the membership of the compute to the backend of the public LB. We need to keep the headless SLB to keep the compatibility when nodes are added without machinesets like in UPI/bring your own RHEL.